src/Queue/QueueHandler.php

```diff
@@ -60,78 +73,85 @@ public function __construct(
      */
     public function handleSqs(SqsEvent $event, Context $context): void
     {
+        $worker = $this->container->makeWith(Worker::class, [
+            'isDownForMaintenance' => fn () => MaintenanceMode::active(),
+        ]);
+
         foreach ($event->getRecords() as $sqsRecord) {
-            $recordData = $sqsRecord->toArray();
-
-            $jobData = [
-                'MessageId' => $recordData['messageId'],
-                'ReceiptHandle' => $recordData['receiptHandle'],
-                'Attributes' => $recordData['attributes'],
-                'Body' => $recordData['body'],
-            ];
-
-            $job = new SqsJob(
-                $this->container,
-                $this->sqs,
-                $jobData,
+            $timeout = $this->calculateJobTimeout($context->getRemainingTimeInMillis());
+
+            $worker->runSqsJob(
+                $job = $this->marshalJob($sqsRecord),
                 $this->connection,
-                $this->queue,
+                $this->gatherWorkerOptions($timeout),
             );
 
-            $this->process($this->connection, $job);
+            if (! $job->hasFailed() and ! $job->isDeleted()) {
+                $job->delete();
+            }
```

Comment:
We immediately remove job after they have been successfully processed, so that if later job from the same batch times out, they won't be retried when the visibility timeout expires.